### ISSUE TITLE
Results on the paper related to mAP seem to be unfair wrt the state of the art 

### ISSUE THREAD

**Issue by mive93** [xingyizhou/CenterNet#612]
I have serious doubts regarding Figure 1 in your paper [1]
In that chart, the mAP of Centernet is compared to the mAP of other neural networks for object detection. 
I have tested and worked on your code and I have noticed that you compute the mAP for your networks with a threshold on the bounding boxes' confidence greater than 0.1. 

Given that, which are the threshold chosen for the other networks (FasterRCNN, RetinaNet, and Yolov3)? Are you sure that it's the same for all of them? Because this threshold seriously affects the value of the mAP and having different values among the compared networks would be unfair and would invalidate your chart. 
Always related to that chart, is the input size that you chose to train your NN the same chosen by the others? This would also change the results. 

Moreover, why are you using a threshold of 0.1 to compute the mAP and a threshold of 0.3 to show the detections? 

Thank you in advance. 
Micaela and @sapienzadavide 

[1] Zhou, Xingyi, Dequan Wang, and Philipp Krähenbühl. "Objects as points." arXiv preprint arXiv:1904.07850 (2019).

**Comment by xingyizhou** [comment]
Thanks for your challenge. 

We didn't use a score threshold (The 0.1 threshold is only for 3D detection) for evaluating mAP, but use the top 100 scored bounding box. Please note that the mAP metric will not penalize bounding boxes in a lower score, so usually no score threshold would give the best results. FasterRCNN, and RetinaNet did use a threshold (if I remembered correctly, 0.05), this is because they need the threshold to perform soft-NMS. I have tested using score threshold 0 on their model but didn't observe better results. 

We did use different input resolutions in our method and FasterRCNN, they give the different speed-accuracy trade-off. The optimal input resolution from different methods should be different, for example, anchor-based methods usually need a larger input size to ensure the anchors overlap sufficiently with ground truth bounding boxes. 

Visualizing using threshold 0 would be ugly.

**Comment by sapienzadavide** [comment]
Thank you very much for your reply. 

Micaela and I have some other questions:

- If you train, for example, CenterNet-DLA34 on 512x512 images and YoloV3 on 416x416 images, the two networks will have different resolution and hence they will have different precisions. Is it correct? I think that it is true also for different networks. 
- The choice to use the first 100 detections (once sorted) seems arbitrary. If you consider a different number of detections for each image, are you sure that your mAP values would be still the same?  
- Using 100 detections for image, on COCO validation set 2017, means that you consider near 500'000 total detections to compute the mAP. On the other hand YoloV3 considers "only" 30'000 detections on COCO validation set. 
- In mAP metrics a big number of false positive seems to have a low weight. Did you also note that?      In case, what is your explanation? Have you noticed that the overall precision (that is just TP/(TP+FP) over all classes and all images) is near only 5%?
- Which threshold would you recommend for a real application, for which false positive are a problem but an high mAP is needed? Because with threshold 0.3 the mAP downgrade to ~34% (using DLA34).

Thank you in advance for you time.
Best regards,
Davide and @mive93

**Comment by xingyizhou** [comment]
Thank you for your questions.
  1. Correct.
  2. I am not sure. The top 100 is from the maximum allowed outputs on COCO evaluation. Feel 
free to play with other top_Ks.
  3. Again, feel free to change top_k to 30 to see how much difference it is.
  4. I am aware of that and this is designed by the mAP metric. In this case setting a score threshold can help to improve TP/(TP+FP).
  5. It really depends on the datasets. COCO has many small objects, which usually come with lower confidence and requires a lower threshold.

**Comment by xingyizhou** [comment]
Closing this for now. Feel free to reopen.
